src/els_parser.erl

```diff
@@ -25,46 +25,47 @@ parse(Text) ->
 
 -spec parse_file(file:io_device()) -> {ok, [poi()]}.
 parse_file(IoDevice) ->
-  {ok, Forms} = els_dodger:parse(IoDevice, {1, 1}),
-  Tree = erl_syntax:form_list(Forms),
-  %% Reset file pointer position.
-  {ok, 0} = file:position(IoDevice, 0),
-  {ok, Extra} = parse_attribute_pois(IoDevice, [], {1, 1}),
+  {ok, NestedPOIs} = els_dodger:parse(IoDevice, {1, 1}, fun parse_form/3, []),
   ok = file:close(IoDevice),
-  POIs = points_of_interest(Tree),
-  {ok, Extra ++ POIs}.
+  {ok, lists:flatten(NestedPOIs)}.
 
-%%==============================================================================
-%% Internal Functions
-%%==============================================================================
+-spec parse_form(file:io_device(), any(), [any()]) ->
+    {'ok', erl_syntax:forms()
+  | none, integer()}
+  | {'eof', integer()}
+  | {'error', any(), integer()}.
+parse_form(Dev, L0, Options) ->
+  parse_form(Dev, L0, fun els_dodger:normal_parser/2, Options).
 
--spec parse_attribute_pois(io:device(), [poi()], erl_anno:location()) ->
-   {ok, [poi()]} | {error, any()}.
-parse_attribute_pois(IoDevice, Acc, StartLoc) ->
-  case io:scan_erl_form(IoDevice, "", StartLoc) of
-    {ok, Tokens, EndLoc} ->
-      case erl_parse:parse_form(Tokens) of
-        {ok, Form} ->
-          POIs = find_attribute_pois(Form, Tokens),
-          parse_attribute_pois(IoDevice, POIs ++ Acc, EndLoc);
-        {error, _Error} ->
-          parse_attribute_pois(IoDevice, Acc, EndLoc)
+-spec parse_form(file:io_device(), any(), function(), [any()]) ->
+  {'ok', erl_syntax:forms() | none, integer()}
+  | {'eof', integer()}
+  | {'error', any(), integer()}.
+parse_form(Dev, L0, Parser, _Options) ->
```

Comment:
Can we please use the original naming for the variables? E.g. `IoDevice` instead of `Dev`, `StartLoc` instead of `L0`, `Tokens` instead of `Ts`, etc?